src/test/java/de/dhbw/karlsruhe/students/mailflow/unit/FileMailboxRepositoryTest.java

```diff
@@ -65,36 +67,29 @@ public Mailbox deserializeMailboxFile(File mailboxFile) {
 
   @ParameterizedTest(name = "should create directories and files for mailboxType {0}")
   @EnumSource(MailboxType.class)
-  void searchingNonExistentMailboxShouldCreate(MailboxType mailboxType)
+  /**
+   * #BugFix
+   */
+  void searchingNonExistentMailboxShouldCreateParsableAndEmptyMailbox(MailboxType mailboxType)
       throws MailboxSavingException, MailboxLoadingException {
     // Arrange
     Address mailboxOwner = new Address("someOwner", "someDomain.de");
     Mailbox searchingMailbox = Mailbox.create(mailboxOwner, Map.of(), mailboxType);
     File directoryOfUser = new File(allMailboxesDirectory, mailboxOwner.toString());
     File mailboxFile = new File(directoryOfUser, mailboxType.getStoringName() + ".json");
 
-    MailboxConverter mockedMailboxConverter =
-        new MailboxConverter() {
-          @Override
-          public String serializeMailbox(Mailbox mailbox) {
-            // not tested
-            return "someSerializedMailboxJson";
-          }
-
-          @Override
-          public Mailbox deserializeMailboxFile(File mailboxFile) {
-            return searchingMailbox;
-          }
-        };
     this.fileMailboxRepository =
-        new FileMailboxRepository(mockedMailboxConverter, allMailboxesDirectory);
+        new FileMailboxRepository(new JSONMailboxConverter(), allMailboxesDirectory);
```

Comment:
No this test is not atomic anymore. It now depends on a correct implementation of the JSONMailboxConverter. Maybe you can use the mock again with a correct serialized string instead? Otherwise you could move this outside of a unit test